web/client/utils/QueryParamsUtils.js

```diff
@@ -222,28 +231,97 @@ export const paramActions = {
             })
         ];
     },
-    featureinfo: (parameters, state) => {
-        const value = parameters.featureinfo;
+    featureInfo: (parameters, state) => {
+        const value = parameters.featureInfo;
         const {lat, lng, filterNameList} = value;
+        const projection = mapProjectionSelector(state);
         if (typeof lat !== 'undefined' && typeof lng !== 'undefined') {
-            const projection = mapProjectionSelector(state);
             return [featureInfoClick(updatePointWithGeometricFilter({
                 latlng: {
                     lat,
                     lng
                 }
             }, projection), false, filterNameList ?? [])];
+        } else if (typeof value === 'string') {
+            const [latitude, longitude] = value.split(',');
+            if (typeof latitude !== 'undefined' && typeof longitude !== 'undefined') {
+                return [featureInfoClick(updatePointWithGeometricFilter({
+                    latlng: {
+                        lat: latitude,
+                        lng: longitude
+                    }
+                }, projection), false, [])];
+            }
         }
         return [];
     },
-    zoom: () => {
+    mapInfo: (parameters) => {
+        const value = parameters.mapInfo;
+        const filterValue = parameters.mapInfoFilter;
+        if (typeof value === 'string') {
+            // use delayed action dispatching if we have same layer name for mapInfo parameter and addLayers parameter
+            const layers = parameters.addLayers;
+            if (typeof layers === 'string') {
+                const parsed = layers.split(',');
+                const pairs = parsed.map(el => el.split(";"));
+                if (pairs.find(el => el[0] === value)) {
+                    return [
+                        scheduleSearchLayerWithFilter({layer: value, cql_filter: filterValue ?? ''})
+                    ];
+                }
+            }
+            return [
+                searchLayerWithFilter({layer: value, cql_filter: filterValue ?? ''})
+            ];
+        }
+        return [];
     },
-    heading: () => {
+    addLayers: (parameters, state) => {
+        const layers = parameters.addLayers;
+        if (typeof layers === 'string') {
+            const parsed = layers.split(',');
+            if (parsed.length) {
+                const defaultSource = selectedServiceSelector(state);
+                const pairs = parsed.map(el => el.split(";"));
+                const layerFilters = (parameters.layerFilters ?? '').split(',') ?? [];
```

Comment:
we should use ";". "," is reserved in cql.

Reply:
Thanks, updated it